src/handlers/issue-deduplication.ts

```diff
@@ -30,16 +77,46 @@ export async function issueChecker(context: Context<"issues.opened" | "issues.ed
     octokit,
     payload,
   } = context;
-  const issue = payload.issue;
-  let issueBody = issue.body;
+
+  const originalIssue = payload.issue;
+
+  // Wait for the configured timeout period
+  await sleep(context.config.editTimeout, context);
+
+  const currentEventTime = context.eventName === "issues.opened" ? new Date(originalIssue.created_at).getTime() : new Date(originalIssue.updated_at).getTime();
+
+  const lastEditedAt = await fetchLastEditTime(context, originalIssue.node_id);
+
+  if (lastEditedAt === null) {
+    logger.info("Last edited time is null, skipping deduplication check", {
+      issueNumber: originalIssue.number,
+    });
+    return;
+  }
+
+  const latestEventTime = new Date(lastEditedAt).getTime();
+
+  // Event should only proceed if it's the most recent action
```

Comment:
This is interesting how it's implemented have you tested that this works as expected?

Reply:
> This is interesting how it's implemented have you tested that this works as expected?

It does work. [Issue](https://github.com/ShivTestOrg/test-public/issues/257),  it skip issue-dedupe if its not the latest event specifically on the issue spec.